### ISSUE TITLE
RMG crash

### ISSUE THREAD

**Issue by kambala-decapitator** [vcmi/vcmi#1377]
**Describe the bug**
RMG crashes when trying to create a map and client is back to menu, reproduced 2 times in a row.

**Game logs**
[VCMI_Server_log.txt.zip](https://github.com/vcmi/vcmi/files/10357776/VCMI_Server_log.txt.zip)

a bit longer stack trace that I received on first attempt:

```
INFO global [0x700000ee8000] - Starting zone placement
ERROR global [0x700000ee8000] - Error: signal 11 :
ERROR global [0x700000ee8000] - 0   vcmiserver                          0x00000001035972f7 _Z17handleLinuxSignali + 39
ERROR global [0x700000ee8000] - 1   libsystem_platform.dylib            0x00007fff204c5d7d _sigtramp + 29
ERROR global [0x700000ee8000] - 2   ???                                 0x0000000000000000 0x0 + 0
ERROR global [0x700000ee8000] - 3   libvcmi.dylib                       0x0000000103ad6822 _ZN11CZonePlacer21attractConnectedZonesERNSt3__13mapIiNS0_10shared_ptrI4ZoneEENS0_4lessIiEENS0_9allocatorINS0_4pairIKiS4_EEEEEERNS1_IS4_6float3NS5_IS4_EENS7_INS8_IKS4_SE_EEEEEERNS1_IS4_fSF_NS7_INS8_ISG_fEEEEEE + 370
ERROR global [0x700000ee8000] - 4   libvcmi.dylib                       0x0000000103ad414d _ZN11CZonePlacer10placeZonesEP16CRandomGenerator + 557
ERROR global [0x700000ee8000] - 5   libvcmi.dylib                       0x0000000103ab88a0 _ZN13CMapGenerator8genZonesEv + 48
ERROR global [0x700000ee8000] - 6   libvcmi.dylib                       0x0000000103ab85e3 _ZN13CMapGenerator8generateEv + 259
ERROR global [0x700000ee8000] - 7   libvcmi.dylib                       0x0000000103c255ed _ZN10CGameState11initNewGameEPK11IMapServiceb + 205
ERROR global [0x700000ee8000] - 8   libvcmi.dylib                       0x0000000103c24e03 _ZN10CGameState4initEPK11IMapServiceP9StartInfob + 387
ERROR global [0x700000ee8000] - 9   vcmiserver                          0x00000001034e7325 _ZN12CGameHandler4initEP9StartInfo + 181
ERROR global [0x700000ee8000] - 10  vcmiserver                          0x00000001035925a5 _ZN11CVCMIServer18prepareToStartGameEv + 837
ERROR global [0x700000ee8000] - 11  vcmiserver                          0x00000001035d826c _ZN14LobbyStartGame13applyOnServerEP11CVCMIServer + 44
ERROR global [0x700000ee8000] - 12  vcmiserver                          0x00000001035b3e5d _ZNK14CApplyOnServerI14Lob
```

**To Reproduce**
Steps to reproduce the behavior: try to create map with settings as on SS

**Expected behavior**
Map is created and started

**Actual behavior**
RMG crashes

**Did it work earlier?**
who knows

**Screenshots**
![Screen Shot 2023-01-06 at 09 14 58](https://user-images.githubusercontent.com/1557784/210942506-56243b4d-446e-421f-bb18-67898f07427c.png)

**Version**
 - OS: macOS Intel
 - Version: develop - battle fixes + HW cursor

**Comment by DjWarmonger** [comment]
Any specific template you have enabled?

**Comment by DjWarmonger** [comment]
Okay, I had this happen **once** on Jebus G+U (but not the other G+U templates).

I suspect some sort of memory shortage or maybe stack overflow. Keep in mind that in game the random map is generated twice.

**Comment by kambala-decapitator** [comment]
> Any specific template you have enabled?

on the SS I have default template selected, don't have any other templates installed other than VCMI extras (Jebus Cross is disabled btw)

**Comment by DjWarmonger** [comment]
Could be anything and everything. I fixed a few (rare) crashes for 1.3.
